### PR TITLE
Closes #8492: Gracefully handle network failures while obtaining token server url

### LINKED ISSUE #8492: Crash in [@ mozilla.appservices.fxaclient.FxaException$Network: at mozilla.appservices.fxaclient.rust.RustError.intoException(RustError.kt:5)]
In the last day, [the rate of crashes with this signature](https://crash-stats.mozilla.org/signature/?product=Fenix&version=0.0a1&signature=mozilla.appservices.fxaclient.FxaException%24Network%3A%20at%20mozilla.appservices.fxaclient.rust.RustError.intoException%28RustError.kt%3A5%29&date=%3E%3D2020-03-24T18%3A42%3A00.000Z&date=%3C2020-09-24T18%3A42%3A00.000Z&_columns=date&_columns=product&_columns=version&_columns=build_id&_columns=platform&_columns=reason&_columns=address&_columns=install_time&_columns=startup_crash&_sort=-build_id&_sort=-date&page=1#graphs) has gone up considerably. Maybe a regression from #7856? @grigoryk 

It sounds similar to #7536 which was about not handling network failures, if I understand things correctly.

Crash report: https://crash-stats.mozilla.org/report/index/92c220aa-3be3-4c80-b071-af3e30200924

Java stack trace:
```
mozilla.appservices.fxaclient.FxaException$Network
	at mozilla.appservices.fxaclient.rust.RustError.intoException(RustError.kt:5)
	at mozilla.appservices.fxaclient.FirefoxAccount.getTokenServerEndpointURL(FirefoxAccount.kt:9)
	at mozilla.components.service.fxa.FirefoxAccount.getTokenServerEndpointURL(FirefoxAccount.kt:1)
	at mozilla.components.service.fxa.manager.FxaAccountManager.maybeUpdateSyncAuthInfoCache(FxaAccountManager.kt:13)
	at mozilla.components.service.fxa.manager.FxaAccountManager$maybeUpdateSyncAuthInfoCache$1.invokeSuspend(Unknown Source:11)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:3)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:21)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
```

### PR DESCRIPTION
**Pull request by grigoryk**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt [hunk 1]
```diff
@@ -184,8 +184,10 @@ class FirefoxAccount internal constructor(
         }
     }
 
-    override fun getTokenServerEndpointURL(): String {
-        return inner.getTokenServerEndpointURL()
+    override suspend fun getTokenServerEndpointURL() = withContext(scope.coroutineContext) {
```
Comment (csadilek) [inline components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt:187]: Curious how this didn't fail before? Was it wrapped in an `handleFxaExceptions` and we lost that? Or are we calling it more often now?

**Review by csadilek — APPROVED** [review]
👍 

Curious to learn more / see below. Also, did you mean to commit the metrics files too?

File: components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt (hunk 1 above)
Comment (grigoryk) [reply components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/FirefoxAccount.kt:187]: Still investigating this. It's neither of those, I'll let you know once I sort this out. Maybe it _was_ failing and we didn't notice, somehow..?

**Comment by grigoryk** [comment]
RE metrics - yup, it keeps getting out of date, somehow, I commit these diffs when I see them locally. Quite annoying.

**Pull request merged**